src/Message/ProceedResponse.php

```diff
@@ -22,51 +22,49 @@ class ProceedResponse extends AbstractResponse
     const PAYMENT_SUCCESS_STATUS_AUTHORIZED = 'AUTHORIZED';
 
     /**
-     * @var string When the proceed request was cancelled (they have a typo in their api)
+     * @var string When the proceed request was cancelled
      */
-    const PAYMENT_SUCCESS_STATUS_CANCELLED = 'CANCELED'; // [sic]
+    const PAYMENT_SUCCESS_STATUS_CANCELLED = 'CANCELED';
 
     /**
      * {@inheritdoc}
      */
     public function isSuccessful()
     {
         if (
-            isset($this->data->proceedSuccess)
+            isset($this->data->proceedSuccess->success->code)
```

Comment:
Is there a possible scenario case when `->success` isn't callable? (See also in the function `isCancelled()`).

Reply:
`$this->data->proceedSuccess` is an `stdClass` and when the property does not exist it triggers a PHP warning, but might carry on executing the code. IMHO It's better to check the existence of the actual property you want to have the value of, instead of just checking the parent.

A little bit of defensive programming. 😉